src/test/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/resource/mapper/induction/FutureWorkInterestsResourceMapperTest.kt

```diff
@@ -13,34 +16,140 @@ class FutureWorkInterestsResourceMapperTest {
+      workInterests = setOf(WorkType.OTHER, WorkType.TECHNICAL),
+      workInterestsOther = "Entertainment",
+      particularJobInterests = setOf(
+        WorkInterestDetail(
+          workInterest = WorkType.OTHER,
+          role = "Juggler",
+        ),
+        WorkInterestDetail(
+          workInterest = WorkType.TECHNICAL,
+          role = "Kotlin Developer",
+        ),
+      ),
+    )
+    val expectedInterests = listOf(
+      WorkInterest(
+        workType = WorkInterestType.OTHER,
+        workTypeOther = "Entertainment",
+        role = "Juggler",
+      ),
+      WorkInterest(
+        workType = WorkInterestType.TECHNICAL,
+        workTypeOther = null,
+        role = "Kotlin Developer",
+      ),
+    )
 
     // When
     val actual = mapper.toCreateFutureWorkInterestsDto(request, prisonId)
 
     // Then
     assertThat(actual!!.prisonId).isEqualTo(prisonId)
-    assertThat(actual.interests).hasSize(1)
-    assertThat(actual.interests[0].workType).isEqualTo(WorkInterestType.OTHER)
-    assertThat(actual.interests[0].workTypeOther).isEqualTo("Any job I can get")
-    assertThat(actual.interests[0].role).isEqualTo("Any role")
+    assertThat(actual.interests).usingRecursiveComparison().ignoringCollectionOrder().isEqualTo(expectedInterests)
   }
 
   @Test
   fun `should map to UpdateFutureWorkInterestsDto`() {
     // Given
     val prisonId = "BXI"
-    val request = aValidUpdateWorkInterestsRequest()
+    val request = aValidUpdateWorkInterestsRequest(
+      workInterests = setOf(WorkType.OTHER, WorkType.TECHNICAL),
+      workInterestsOther = "Entertainment",
+      particularJobInterests = setOf(
+        WorkInterestDetail(
+          workInterest = WorkType.OTHER,
+          role = "Juggler",
+        ),
+        WorkInterestDetail(
+          workInterest = WorkType.TECHNICAL,
+          role = "Kotlin Developer",
+        ),
+      ),
+    )
+    val expectedInterests = listOf(
+      WorkInterest(
+        workType = WorkInterestType.OTHER,
+        workTypeOther = "Entertainment",
+        role = "Juggler",
+      ),
+      WorkInterest(
+        workType = WorkInterestType.TECHNICAL,
+        workTypeOther = null,
+        role = "Kotlin Developer",
+      ),
+    )
 
     // When
     val actual = mapper.toUpdateFutureWorkInterestsDto(request, prisonId)
 
     // Then
     assertThat(actual!!.prisonId).isEqualTo(prisonId)
     assertThat(actual.reference).isEqualTo(request.id)
-    assertThat(actual.interests).hasSize(1)
-    assertThat(actual.interests[0].workType).isEqualTo(WorkInterestType.OTHER)
-    assertThat(actual.interests[0].workTypeOther).isEqualTo("Any job I can get")
-    assertThat(actual.interests[0].role).isEqualTo("Any role")
+    assertThat(actual.interests).usingRecursiveComparison().ignoringCollectionOrder().isEqualTo(expectedInterests)
+  }
+
+  @Test
+  fun `should map to UpdateFutureWorkInterestsDto with missing particular job interest`() {
```

Comment:
From the test name it sounds like you're expecting to have deleted a job, but actually you've added one.
How about `should map to UpdateFutureWorkInterestsDto given additional work interests type` ?

Reply:
What makes you say I've "added" one? This is simply mapping from one object (API) to another (domain). There's not yet any notion of what already exists (that's done at the entity layer).

In my mind, two `workInterests` have been provided, but not two `particularJobInterests`, so one of the latter is missing :)